lib/devise/models/authenticatable.rb

```diff
@@ -253,6 +253,7 @@ def find_or_initialize_with_error_by(attribute, value, error=:invalid) #:nodoc:
 
         # Find or initialize a record with group of attributes based on a list of required attributes.
         def find_or_initialize_with_errors(required_attributes, attributes, error=:invalid) #:nodoc:
+          attributes = attributes.to_unsafe_h.with_indifferent_access if attributes.respond_to? :to_unsafe_h
```

Comment:
I think resource_params definitely needs to be sanitized - otherwise it requires calling the to_unsafe_h on the parameters which just doesn't sound like a great idea.  If the relevant fields are permitted in resource_params (have the relevant controllers call the correct sanitizer) then it's just a matter of calling .to_h  on the ActionController::Parameters which returns the filtered HWA